pipelines/jenkins/JenkinsAnchoreCICD.txt

```diff
@@ -0,0 +1,98 @@
+pipeline {
+        agent any
+        environment {
+            //Define some reusable variables here 
+            IMAGENAME="kaizencont/goutil"
+            WAIT_TIMEOUT=10
+            ANCHORE_URL="http://10.0.0.15:8080/v1/"
+            dockerImage = '' 
+            registryCredential = 'dockerhub'
+        }
+        stages {
+             stage ('Build') {
+                steps {
+                    println ("Building image for configured project...")
+                    
+                    // Performs a checkout of the project from the specific git repository
+                    git 'https://github.com/jeremybryan/go-aws-utils.git/'
+                
+                    // Perform a docker build of the project assumes the Dockerfile is located in the project's root directory
+                    script { 
+                            dockerImage = docker.build IMAGENAME+":"+ TAG
+                    }
+                
+                    // Push the created image to the registry configured with registryCredential
+                    script {
+                        docker.withRegistry( '', registryCredential ) { 
+                                dockerImage.push() 
+                        }
+                    }
+                }
+            }
+            
+            stage("Verifying Anchore Availability") {
+                steps {
+                    script {
+                        // We attempt to discover the availability of the Anchore installation. This is currently looking to see if the api endpoint is reachable. If the call fails
+                        // entirely, we catch the exception and log it, but proceed on. We could add additional checking here on http status or other pieces of the returned json object.
+                        //
+                        // Important point here is anchoreStatus is getting set here so we can guard calls to anchore below and not kill the pipeline if Anchore is unreachable.
+                        try {
+                            withCredentials([usernamePassword(credentialsId: 'Anchore-Pass', passwordVariable: 'pass', usernameVariable: 'user')]) {
+                                env.anchoreStatus = sh(script: 'docker run --name anchorecli --rm anchore/anchore-cli-dev anchore-cli --u $user --p $pass --url $ANCHORE_URL --json system status | jq -r \'.service_states[] | select(.servicename == "apiext") .status_message\'', returnStdout: true).trim()
```

Comment:
Should we use `anchore/anchore-cli-dev` or `anchore/engine-cli`? Did you ever find out a difference?

Reply:
I have it on my backlog to go back through a do this with `anchore\engine-cli` to determine if there are differences. I was also planning to do a variant where `anchorectl` is used to the extent it can be.

Reply:
Gotcha yeah that's a good idea. It'll be great to get all these in real examples we can send to people